### PR TITLE
centurion: add package

### PR DESCRIPTION
**Pull request by ilobilo**
* Before adding new features and new modules, please go to issues to submit the relevant feature description first.
* Write good commit messages and use the same coding conventions as the rest of the project.
* Please commit code to dev branch and we will merge into master branch in feature
* Ensure your edited codes with four spaces instead of TAB.

------

* 增加新特性和新模块之前，请先到issues提交相关特性说明，经过讨论评估确认后，再进行相应的代码提交，避免做无用工作。
* 编写友好可读的提交信息，并使用与工程代码相同的代码规范，代码请用4个空格字符代替tab缩进。
* 请提交代码到dev分支，如果通过，我们会在特定时间合并到master分支上。
* 为了规范化提交日志的格式，commit消息，不要用中文，请用英文描述。

### CODE REVIEW

**Comment by ilobilo** [comment]
what can I do about failing checks?

**Comment by star-hengxing** [comment]
https://github.com/xmake-io/xmake-repo/blob/a1a5ca1ff583456ed9519703bdcdf00ce3fa2c0b/packages/l/libsdl/xmake.lua#L249-L255

**Comment by ilobilo** [comment]
you mean I should add ``SDL_MAIN_HANDLED`` to tests?

**Comment by ilobilo** [comment]
oh oops, I copied the function from some package

**Comment by Chi-EEE** [comment]
maybe add:
```lua
if is_plat("wasm") then
    add_configs("shared", {description = "Build shared library.", default = false, type = "boolean", readonly = true})
end
```
since the other libsdl packages have it

File: packages/c/centurion/xmake.lua [hunk 1]
```diff
@@ -60,7 +59,7 @@ package("centurion")
 
     on_test(function (package)
         assert(package:check_cxxsnippets({ test = [[
-            void test(int args, char **argv) {
+            int main() {
```
Comment (waruqi) [inline packages/c/centurion/xmake.lua:62]: please use `void test()`

Comment (ilobilo) [reply packages/c/centurion/xmake.lua:62]: I had ``void test`` but some checks failed so @star-hengxing changed it https://github.com/xmake-io/xmake-repo/commit/ffe9235e3e8038b7fe67d36bd323da0e2db9bcb0

**Comment by ilobilo** [comment]
> maybe add:
> 
> ```lua
> if is_plat("wasm") then
>     add_configs("shared", {description = "Build shared library.", default = false, type = "boolean", readonly = true})
> end
> ```
> 
> since the other libsdl packages have it

done

**Comment by ilobilo** [comment]
what does it want now

**Comment by star-hengxing** [comment]
@c8ef https://github.com/xmake-io/xmake-repo/commit/e167e3a9cc6d83706c50908306f7227c1d47f119 break android ci :(

**Comment by waruqi** [comment]
we can use io.replace or add_patches to fix it.

**Pull request merged**